### PR TITLE
Template deprecation warnings suppression

### PR DESCRIPTION
**Pull request by rsvoboda**
Template deprecation warnings suppression.

Template class is deprecated, there is some logic to do the fallback to Recorder behavior.

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java [hunk 1]
```diff
@@ -530,7 +531,7 @@ private static boolean isRecorder(AnnotatedElement element) {
             if (isRecorder) {
                 boolean recorderFound = false;
                 for (Class<?> p : method.getParameterTypes()) {
-                    if (p.isAnnotationPresent(Recorder.class) || p.isAnnotationPresent(Template.class)) {
```
Comment (kenfinnigan) [inline core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java]: How is this change related to suppressing deprecation warnings?

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java]: It's related because it uses the `isRecorder` method above which has the suppress warning.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks!
